services/net/folder-collection/FolderCollectionManager.cs

```diff
@@ -64,20 +78,63 @@ public class FolderCollectionManager : ServiceManager<FolderCollectionOptions>
         IKafkaAdmin kafkaAdmin,
         IKafkaListener<string, IndexRequestModel> consumer,
         IApiService api,
+        IChesService chesService,
+        IOptions<ChesOptions> chesOptions,
         IOptions<FolderCollectionOptions> options,
         ILogger<FolderCollectionManager> logger)
         : base(api, options, logger)
     {
         this.Client = elasticClient;
         this.ElasticOptions = elasticOptions.Value;
         this.KafkaAdmin = kafkaAdmin;
+        this.Ches = chesService;
+        this.ChesOptions = chesOptions.Value;
         this.Listener = consumer;
         this.Listener.OnError += ListenerErrorHandler;
         this.Listener.OnStop += ListenerStopHandler;
     }
     #endregion
 
     #region Methods
+    /// <summary>
+    /// Send email alert of failure.
+    /// </summary>
+    /// <param name="subject"></param>
+    /// <param name="message"></param>
+    /// <returns></returns>
+    public async Task SendEmailAsync(string subject, string message)
+    {
+        if (this.Options.SendEmailOnFailure)
+        {
+            try
+            {
+                var email = new TNO.Ches.Models.EmailModel(this.ChesOptions.From, this.Options.EmailTo, subject, message);
+                await this.Ches.SendEmailAsync(email);
+            }
+            catch (ChesException ex)
```

Comment:
Does the ChesService ever throw any other type of error?  If it does you'll need to handle it, or this process can break other processes.

Reply:
No, we can remove this PR.